codec/dagjson/marshal.go

```diff
@@ -19,8 +19,11 @@ func Marshal(n ipld.Node, sink shared.TokenSink, allowLinks bool) error {
 	var tk tok.Token
 	switch n.Kind() {
 	case ipld.Kind_Invalid:
-		return fmt.Errorf("cannot traverse a node that is absent")
+		return fmt.Errorf("invalid node encountered")
 	case ipld.Kind_Null:
+		if n.IsAbsent() {
+			return fmt.Errorf("cannot encode a node that is absent")
```

Comment:
i worry that this error message, which will be at runtime, will be fairly opaque. Consider adding a hint like "perhaps you are trying to encode the non-representation version of a node?"